lib/gallery_saver.dart

```diff
@@ -1,41 +1,80 @@
 import 'dart:async';
+import 'dart:io';
 
 import 'package:flutter/services.dart';
+import 'package:gallery_saver/files.dart';
+import 'package:http/http.dart' as http;
+import 'package:path_provider/path_provider.dart';
+import 'package:path/path.dart';
 
 class GallerySaver {
   static const String channelName = 'gallery_saver';
   static const String methodSaveImage = 'saveImage';
   static const String methodSaveVideo = 'saveVideo';
 
-  static const String pleaseProvidePath = 'Please provide valid file path';
-
+  static const String pleaseProvidePath = 'Please provide valid file path.';
+  static const String fileIsNotVideo = 'File on path is not a video.';
+  static const String fileIsNotImage = 'File on path is not an image.';
   static const MethodChannel _channel = const MethodChannel(channelName);
 
   ///saves video from provided temp path
   static Future<bool> saveVideo(String path) async {
+    File tempFile;
     if (path == null || path.isEmpty) {
       throw ArgumentError(pleaseProvidePath);
     }
-
+    if (!isVideo(path)) {
```

Comment:
do we need to check this?

Reply:
Methods save files at videos/photos folders so it is better for us to handle this case here, otherwise files will be stored and won't be visible.